src/main/java/jenkins/branch/Branch.java

```diff
@@ -153,7 +153,7 @@ public String getName() {
      * @since 0.2-beta-7
      */
     public String getEncodedName() {
-        return Util.rawEncode(getName());
+        return NameMangler.apply(getName());
```

Comment:
The main concern I have about this is that it means the `Branch.name` is no longer reconstructible from the `Job.name`. So long as nobody is trying to do that, and is using `BranchProjectFactory.getBranch` correctly, then I guess it does not matter. But you must be sure that there can be no (accidental or malicious) collision between `encodedName`s.